src/search.js

```diff
@@ -1,23 +1,25 @@
-import {assoc, concat, filter, has, isEmpty, map, pipe, prop, sortBy} from "ramda"
+import {ascend, assoc, concat, descend, has, isEmpty, map, prop, sortBy, sortWith} from "ramda"
 import * as diacritics from 'diacritics'
 
 
 export function findParametersAndVariables(parameters, variables, query) {
-  function matchesQuery(parameterOrVariable) {
-    if (isEmpty(normalizedQuery)) {
-      return true
-    }
-    return parameterOrVariable.name.includes(normalizedQuery) ||
-      parameterOrVariable.searchIndex.includes(normalizedQuery)
+  function indexInString(substring, string) {
+    const index = string.indexOf(substring)
+    return index === -1 ? Number.MAX_SAFE_INTEGER : index
   }
   const normalizedQuery = diacritics.remove(query.toLowerCase())
-  return pipe(
-    filter(matchesQuery),
-    sortBy(prop('name')),
-  )(concat(
+  let parametersAndVariables = concat(
     map(assoc('type', 'parameter'), parameters),
     map(assoc('type', 'variable'), variables),
-  ))
+  )
+  const sortFunction = isEmpty(normalizedQuery)
+    ? sortBy(prop('name'))
+    : sortWith([
+      ascend(item => indexInString(normalizedQuery, item.name)),
+      descend(item => item.searchIndex.includes(normalizedQuery)),
+      ascend(prop('name')),
+    ])
+  return sortFunction(parametersAndVariables)
 }
 
 export function withSearchIndex(propertyName, objects) {
```

Comment:
This is _really_ hard to understand. Please at least rename the function so that it has an active verb that makes its action clear, and the arguments so that one can get an idea of what type of objects and property names are expected to be passed.
What I get is that you add a `searchIndex` property to each object in an array of objects, which can be either an empty string (then why the hell is it called `searchIndex`? I would expect a Number in there!) or a String taken from some property of that object with removed diacritics. It takes me 5 minutes to decode that algorithm and I don't understand the intention at all  😞